pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -269,17 +261,15 @@ impl InMemoryLayer {
     /// Adds the page version to the in-memory tree
     pub async fn put_value(&self, key: Key, lsn: Lsn, val: &Value) -> Result<()> {
         trace!("put_value key {} at {}/{}", key, self.timeline_id, lsn);
-        let mut inner = self.inner.write().await;
+        let inner: &mut _ = &mut *self.inner.write().await;
         self.assert_writable();
 
         let off = {
-            SER_BUFFER.with(|x| -> Result<_> {
-                let mut buf = x.borrow_mut();
-                buf.clear();
-                val.ser_into(&mut (*buf))?;
-                let off = inner.file.write_blob(&buf)?;
-                Ok(off)
-            })?
+            // Avoid doing allocations for "small" values.
+            let mut buf = smallvec::SmallVec::<[u8; 128]>::new();
+            buf.clear();
+            val.ser_into(&mut buf)?;
+            inner.file.write_blob(&buf).await?
         };
```

Comment:
Is this 128 magic number chosen based on some deep insight or just shot from the hip? :) 

I guess we'd want to cover the largest `Value::WalRecord(NeonWalRecord)` variant?

Reply:
I just took a guess at how large values typically are in databases, and I'd wager most should be below. Note that we care about the deep size here of the values, as we serialize them, so even though `Value::Image` only stores four pointers, its content can be way larger.

If you have any data about the distribution of the value sizes, that would be very useful.

Reply:
I ran some tests, according to them a limit of 128 means we don't allocate in three quarters of cases. See [above](https://github.com/neondatabase/neon/pull/5056#discussion_r1301975061).

Reply:
> 3 bytes overhead
This is why I was thinking about the alternative ways to write and not copy everything to place A and then (in parts) to B.